From 05abbb3dc82d937a5c68d7ffcbea578e0dff58ed Mon Sep 17 00:00:00 2001
From: Debasish Biswas <debasishbsws.dev@gmail.com>
Date: Wed, 15 Jan 2025 00:14:35 +0530
Subject: [PATCH] Fix BUILD v4.0.0: Update Boost API usage: Replace reset() with restart()

In the latest version of boost 1.87 the API change the upstream uses the older API to address that replace the deprecated ioContext.reset() with ioContext.restart().

Signed-off-by: Debasish Biswas <debasishbsws.dev@gmail.com>
---

diff --git a/common/ChildProcess/IoContext.cpp b/common/ChildProcess/IoContext.cpp
index 56f1422..490c708 100644
--- a/common/ChildProcess/IoContext.cpp
+++ b/common/ChildProcess/IoContext.cpp
@@ -48,7 +48,7 @@ struct IoContext::Impl
                 while (!cancellationToken.stop_requested())
                 {
                     ioContext.run_for(boost::asio::chrono::milliseconds(500));
-                    ioContext.reset();
+                    ioContext.restart();
                 }
             }
             catch (std::exception const &e)
